### DISCUSSION TITLE
RSA example doesn't compile

### DISCUSSION THREAD

**Discussion by leo150** [krzyzanowskim/CryptoSwift#980] [Q&A] [answered]
Getting an error while trying to use methods that are declared public with an extension. 
I use main branch to get RSA support.

Steps to reproduce the issue:
1. Create fresh iOS project
2. Add CryptoSwift pod: `pod 'CryptoSwift', :git => 'git@github.com:krzyzanowskim/CryptoSwift.git'`
3. Try to use any "public" method `let rsa = try? RSA(rawRepresentation: Data())`
4. Get a random error: `Incorrect argument label in call (have 'rawRepresentation:', expected 'keySize:')`

Using other methods that were declared public with [an extension ](https://github.com/krzyzanowskim/CryptoSwift/blob/bee7ed5f2d4488dd3df4e4251aaeb64451c106ec/Sources/CryptoSwift/PEM/DER.swift#L80) has the same result:
```
let rsa = try? RSA(keySize: 0)
try? rsa.externalRepresentation() // No exact matches in call to instance method 'externalRepresentation'
```

Might be related to Swift 5.7

Is there any workaround? Thanks

**Comment by leo150** [comment]
Looks like it's [this](https://github.com/krzyzanowskim/CryptoSwift/commit/d18b326c84e46232b0ad5745b18c309e1f994cde) commit that made `DEREncodable` internal so public methods aren't visible anymore. 
@btoms20 could you please take a look?

**Reply by btoms20** [reply]
Thanks for catching this! Not sure why I switched everything back to `internal`. I'm more than happy to make the `DEREncodable` and `DERDecodable` protocols and their associated methods `public` again. 

@NathanFallet, do you think we should open up the entirety of the `DERCodable` API or should we selectively open some methods and initializers (in order to keep the public facing API cleaner and more straight forward)?

``` swift
/// Keep these internal 
internal init(publicDER: Array<UInt8>) throws
internal init(privateDER: Array<UInt8>) throws

/// Make this public
public init(rawRepresentation: Data) throws
```

Same with the exports...
``` swift
/// Keep these internal
internal func publicKeyDER() throws -> Array<UInt8>
internal func privateKeyDER() throws -> Array<UInt8>

/// Make these public
public func externalRepresentation() throws -> Data
public func publicKeyExternalRepresentation() throws -> Data
```

Let me know what you prefer and I'll draft up a PR...

**Reply by nathanfallet** [reply]
The main goal of the library is to expose cryptographic features. However, this protocol cab be useful to be used. So making it public allows it to be used.
Mention me on the PR and I'll be happy to merge it!

**Comment by btoms20** [comment, marked as answer]
@leo150 - #981 should fix this 👍

**Reply by leo150** [reply]
Thank you!

**Comment by bemisal01** [comment]
Hi everyone 
 I am trying to make connection with smart tv but getting ssl handshake failure error because I don't have a purchased certificate for it . can anyone help me out how I can fix this without purchasing any ssl certificate . can this library useful and how I can use it to resolve my issue . 
![IMG_0645 HEIC](https://user-images.githubusercontent.com/110657800/194057801-406e5ec0-ba44-4a14-9ad6-dda2cccae86c.jpg)
